l10n_th_withholding_tax_multi/models/account_payment_register.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = "account.payment.register"
+
+    @api.depends("amount")
+    def _compute_payment_difference(self):
+        super()._compute_payment_difference()
+        for rec in self:
+            if float_compare(rec.payment_difference, 0.0, 2) == 0:
+                rec.payment_difference_handling = "open"
+
+    def _update_payment_register(self, amount_wt, inv_lines):
+        super()._update_payment_register(amount_wt, inv_lines)
+        move_ids = inv_lines.mapped("move_id")
+        amount_residual = sum(move_ids.mapped("amount_residual"))
+        amount_total = sum(move_ids.mapped("amount_total"))
+        if float_compare(amount_residual, amount_total, 2) == 0 and len(inv_lines) > 1:
+            self.payment_difference_handling = "reconcile_multi_deduct"
+            self._onchange_payment_difference_handling()
```

Comment:
I improve this code for use in case of registering a payment with multi invoices.

Waiting for
- [ ] https://github.com/OCA/l10n-thailand/pull/252